libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -114,6 +114,13 @@ class NavigationRoute internal constructor(
         }
     }
 
+    /**
+     * Unique local identifier of the route instance.
+     * For routes which contain server-side UUID it's equal to: `UUID + "#" + routeIndex`, for example: `d77PcddF8rhGUc3ORYGfcwcDfS_8QW6r1iXugXD0HOgmr9CWL8wn0g==#0`.
+     * For routes which were generated onboard and do not have a UUID it's equal to: `"local@" + generateUuid() + "#" + routeIndex`, for example: `local@84438c3e-f608-47e9-88cc-cddf341d2fb1#0`.
+     */
+    val id: String = nativeRoute.routeId
```

Comment:
Sorry, I'm blind, out of current PR scope, but important: need to add `routeIndex` and `id` to equals and hashCode functions, otherwise, they seem collisions prone

Reply:
han, nice suggestion! 👍

Reply:
Done! Also update `toString` to be more readable.

Reply:
Actually had to re-add `directionsRoute` to `equals` because we're changing annotations under the hood without updating the ID and these routes will not be equal anymore.